### ISSUE TITLE
SSHCommandClientEndpoint.newConnection’s 'port' argument does not behave as documented

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6631]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#6631|
|Type|defect|
|Created|2013-07-18 11:53:10Z|
|Branch|https://github.com/twisted/twisted/tree/ssh-endpoint-default-port-6631|

The docs say:

```
The port number of the SSH server. By default, the standard SSH port number is used.
```

This is not true, if the default None is passed, the following happens:

```
Unhandled error in Deferred:
Unhandled Error
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 575, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/base.py", line 253, in _checkTimeout
    userDeferred.callback(result)
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 380, in callback
    self._startRunCallbacks(result)
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 488, in _startRunCallbacks
    self._runCallbacks()
--- <exception caught here> ---
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 575, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/tcp.py", line 440, in _setRealAddress
    self.doConnect()
  File "/home/vagrant/.virtualenvs/webserver_bot/local/lib/python2.7/site-packages/twisted/internet/tcp.py", line 569, in doConnect
    connectResult = self.socket.connect_ex(self.realAddress)
  File "/usr/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
exceptions.TypeError: an integer is required
```

And indeed, I can’t find any logic in the conch code that would take care of this.

The example at http://twistedmatrix.com/documents/current/conch/howto/conch_client.html is not affected because it pulls the port number out of options with a default of 22.

I’ll happily supply a patch unless I’m missing something.

Attachments:

* [SSHCommandClientEndpoint-default-port-6631.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cca/ccaff0fe3eabd25a5db01c96b2e374e6314c6983/9a6cc189fac6ce0c5d3bb37029deec3a0b0d5f4c.patch) (1978 bytes) - added by hynek on 2013-07-18 12:26:37Z - 
* [SSHCommandClientEndpoint-default-port-6631-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cca/ccaff0fe3eabd25a5db01c96b2e374e6314c6983/b5b8caa1ce2fae14b16f35a223107e57e4cb492a.patch) (2276 bytes) - added by hynek on 2013-07-18 12:28:50Z - I fail at git.

<details><summary>Searchable metadata</summary>

```
trac-id__6631 6631
type__defect defect
reporter__hynek hynek
priority__normal normal
milestone__ 
branch__branches_ssh_endpoint_default_port_6631 branches/ssh-endpoint-default-port-6631
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1374148390000000 1374148390000000
changetime__1374218598000000 1374218598000000
version__None None
owner__therve therve
cc__hynek cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_2"></a>|@hynek commented|
|-|-|

Seems to be a real bug. Proposed patch attached.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

(In [39224]) Branching to 'ssh-endpoint-default-port-6631'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [39229]) Merge ssh-endpoint-default-port-6631

Author: hynek
Reviewer: therve
Fixes: [#6631](https://github.com/twisted/twisted/issues/6631)

Use default 22 port in SSHCommandClientEndpoint when no port number is
specified, as documented.
